refactor: trying to improve main overview page of the doc (#651)
ZupIT/beagle-docs: content/en/overview.md

## What is Beagle?

Beagle is an **open source framework** that helps developers implement **Server-Driven UI** in a **cross-platform** way.

By using Beagle, developers can:

{{% alert color="success" %}}
- **Rapidly change an application layout, data, navigation flow, or even logic**, just by changing code in your backend
- **Be more independent of mobile stores** (App Store and Play Store), since most changes won't need an app update
- **Have more confidence that applications behave similar on different platforms**, since more code will be shared and standarized between backend and frontend

> - **Easily test new business hipothesis or make important fixes on live applications** to rapidly improve users experience and receive feedbacks

Reviewer: O que acha de omitirmos um pouco as palavras "you" e "your"? Lendo as docs em inglês da uma canseira de tanto que aparece heheeh

> - **Easily test new business hipothesis or make important fixes on live applications** to rapidly improve users experience and receive feedbacks

Author: If we stick with "you can" on each list item, we should probably change the previous sentence: 

```txt
By using Beagle, developers can: 
- ...
```

> - **Easily test new business hipothesis or make important fixes on live applications** to rapidly improve users experience and receive feedbacks

Author: I can't say about other places on our doc, but here I would say I prefer without using "you can" on each item because it seems redundant. But I can also see that it sounds better on some.

> - **Easily test new business hipothesis or make important fixes on live applications** to rapidly improve users experience and receive feedbacks

Reviewer: Suggestions to avoid you / your:
"You can rapidly change... " => "It's possible to rapidly change..."
"By using Beagle, you can:" => "By using Beagle, it's possible:"
"Put this in your BFF..." => "Put this in the/a BFF"
"You need to configure your build.gradle file" => "Configure the build.gradle file"
"Throughout our documentation, you will see examples..." => "Throughout the documentation, there are examples..."

There's always a way to convert you/your to something else, and I don't think it is necessary to include you/your all the time. And it becomes a little informal, considering the Documentation context.

> - **Easily test new business hipothesis or make important fixes on live applications** to rapidly improve users experience and receive feedbacks

Reviewer 2: Claro, tem outras formas. Vou sugerir outra, por exemplo

You are able to:  
**Change an application layout, data, navigation flow, or even logic**, just change the code in your backend.
- **You don't have to depend on mobile stores** like App Store and Play Store because most changes won't need an app update.
- **Applications will behave similarly on different platforms** since the code will be shared and standardized between backend and frontend.
- **Test a new business hypothesis or make important fixes on live applications** to improve users' experience and receive feedback.

{{% /alert %}}

## How does Beagle work?

> The best way to understand how Beagle works is to see it in action. Here, we will show you a simple screen declared via Beagle, containing just texts, images, and some style configurations. At the end of this section, you will be familiar with Beagle's building blocks and most common features.

Author: @jessica-tw Do you think "The best way to understand Beagle is to see it in action" would be ok?
Because I wanted to sound like we are going to provide an actual example to our readers

> The best way to understand how Beagle works is to see it in action. Here, we will show you a simple screen declared via Beagle, containing just texts, images, and some style configurations. At the end of this section, you will be familiar with Beagle's building blocks and most common features.

Reviewer 2: Ah ok, I got it. I don't see this expression very often, but if you guys are used to it, just keep it. There are other ways to say it too if you want more options: 
- The best way to understand Beagle is to see how it works in a live example. 
- The best way to understand Beagle is to see how it works in action.

> Throughout our documentation, you will see examples which use this tool we call **Playground**. By using it, you can quickly see how Beagle work in real life. You can, as well, edit the code on the left, and select different platforms to run your code on. To better understand how to use this tool, you can check this [section]({{< ref path="playground/_index.md" >}}).

### Components

Now, let's take a closer look inside this JSON, so you can better understand Beagle's capabilites. The first thing to notice is this structure:

```json
{
  "_beagleComponent_": "beagle:container",
  "style": {...},
  "children": [...]
}
```

This is what we call a **Component**, and you can ensure that due to the attribute `_beagleComponent_`. Beagle comes with a bunch of useful components (you can navigate through them later [here]({{< ref path="api/components/_index.md" >}})), and you can also define your own components, which we call **Custom Components**, but that's a later topic. This one is a basic and important component named [Container]({{< ref path="api/components/layout/container.md" >}}), it allows you to *group together* those components inside attribute `children`.

> There are other components with the *children* attribute (or sometimes just *child*) like [Screen]({{< ref path="api/screen/_index.md" >}}) and [ListView]({{< ref path="api/components/layout/listview.md" >}}), and they are usually used to **compose view hierarchies**. In this example, we have 3 other components inside *Container*: [Text]({{< ref path="api/components/ui/text.md" >}}), [Image]({{< ref path="api/components/ui/image/_index.md" >}}), and [Button]({{< ref path="api/components/ui/button.md" >}}). Each one have different attributes you can use to customize their rendering, and you can see all these attributes in their documentation.

Author: Personally, I prefer using ["Oxford's commas"](https://en.wikipedia.org/wiki/Serial_comma), but if that's not the pattern being used throughout our documentation, we can definitely change it here

> The *Image* component, for example, has this attribute named `path` that can receive `remote` or `local` paths to the image data. Here we use a `remote` image by providing an `url` which Beagle will use to request the image when the component gets rendered.

Reviewer: ```suggestion
The *Image* component, for example, has this attribute named `path` that can receive `remote` or `local` paths to the image data. Here we use a `remote` image by providing a `url` which Beagle will use to request the image when the component gets rendered.
```

```json
{
  "_beagleComponent_": "beagle:image",
  "path": {
    "_beagleImagePath_": "remote",
    "url": "https://i.ibb.co/rvRN9kv/logo.png"
  },
  "style": {...}
}
```

> > You can have full control on the network request triggered by this remote image, you just need to configure your own Network Layer as a Beagle Dependency. Beagle have many configurable pieces like this, you can see more in [Customization]({{< ref path="resources/customization/_index.md" >}}).

Reviewer: ```suggestion
> You can have full control on the network request triggered by this remote image, you just need to configure your own Network Layer as a Beagle dependency. Beagle have many configurable fragments like that and you can see more in [Customization]({{< ref path="resources/customization/_index.md" >}}).
```

### Styling

> Finally, let's take a closer look into the `style` attribute, which describes how to position and layout components and their children. Most of components have this attribute, and it's responsible for an important Beagle feature: **developers have control *through backend* on UI positioning**. You can see this power by chaging the `flexDirection` attribute to `ROW` inside the playground, and you will see the same views positioned horizontally. In a real application, you could deploy this exact change in your backend, and they would be *immediately* reflected in your frontend – even on mobile platforms, without needing mobile stores update.

Reviewer: ```suggestion
Finally, let's take a closer look into the `style` attribute, which describes how to position and layout components and their children. Most of components have this attribute, and it's responsible for an important Beagle feature: **developers have control *through backend* on UI positioning**. You can see this feature by changing the `flexDirection` attribute to `ROW` inside [Beagle's Playground](https://playground.usebeagle.io/), and you will see the same views positioned horizontally. In a real application, you could deploy this exact change in your backend, and they would be *immediately* reflected in your frontend – even on mobile platforms, without needing mobile stores update.
```

> Many tools built in-house for Server Driven UI don't allow this kind of power over UI positioning, and this come out of the box with Beagle.

```json
{
  "_beagleComponent_": "beagle:container",
  "style": {
    "flex": {
      "flexDirection": "COLUMN",
      "alignItems": "CENTER",
      "justifyContent": "CENTER"
    },
    "size": {
      "height": {
        "value": 100,
        "type": "PERCENT"
      }
    },
    "backgroundColor": "#FFF"
  },
  "children": [...]
}
```

In our *Container*, we are using 3 styling attributes: `flex`, `size`, and `backgroundColor`. There are other options as well, you can see them all [here]({{< ref path="api/widget.md#style-attributes" >}}).

The `flex` attribute is pretty important because it allows you to **use the same Layout Engine on different platforms**. Since all your platforms will be positioning views according to the same rules, and you don't need to "duplicate" the same layout logic for each platform, this can be a huge win for your team.

> If you are familiar with web development, you probably already know how to use `flex` because it's used as a cross-platform [CSS Flexbox](https://www.w3schools.com/css/css3_flexbox.asp). To accomplish this behind the scenes, we leverage a library called [**Yoga**](https://yogalayout.com), a cross-platform C++ library developed by Facebook and used in other projects (e.g: React Native). If you are not familiar with Flexbox, it takes some time to get used to it, but we bet you will really enjoy it due to its simplicity, power, and universality. [Our documentation]({{< ref path="resources/components-positioning/_index.md" >}}), and [Yoga's own documentation](https://yogalayout.com/docs) can help you to quickly understand it.

Reviewer: ```suggestion
If you are familiar with web development, you probably already know how to use `flex` because it's used cross-platform [CSS Flexbox](https://www.w3schools.com/css/css3_flexbox.asp). To accomplish this behind the scenes, we leverage a library called [**Yoga**](https://yogalayout.com), that is a cross-platform C++ library developed by Facebook and used in other projects (e.g: React Native). If you are not familiar with Flexbox, it takes some time to get used to it, but we bet you will really enjoy it due to its simplicity, power and flexibility. [Our documentation]({{< ref path="resources/components-positioning/_index.md" >}}), and [Yoga's own documentation](https://yogalayout.com/docs) can help you to quickly understand it.
```

### Overview of Beagle's architecture

![An overview of Beagle's architecture](/shared/beaglemobileback.png)

---

## Platforms and Versions

Beagle has different libraries/frameworks for each supported platform, in the following list you can see and access the most up to date versions:

{{% alert color="info" %}}
**Backend Kotlin:** [![back](https://camo.githubusercontent.com/27998a386042ecb2cae7b9f09ae159bd07c935bd/68747470733a2f2f696d672e736869656c64732e696f2f6d6176656e2d63656e7472616c2f762f62722e636f6d2e7a75702e626561676c652f6672616d65776f726b)](https://mvnrepository.com/artifact/br.com.zup.beagle/framework)

**Mobile**:

> - **Android:** [![Maven Central](https://img.shields.io/maven-central/v/br.com.zup.beagle/android)](https://mvnrepository.com/artifact/br.com.zup.beagle/android)
> - **iOS:** [![badge](https://img.shields.io/cocoapods/v/Beagle)](https://cocoapods.org/pods/Beagle)
> - **React Native:** [![react native badge](https://img.shields.io/npm/v/@zup-it/beagle-react-native?logo=React)](https://www.npmjs.com/package/@zup-it/beagle-react-native)
> - **Flutter:** *Being developed in this [folder](https://github.com/ZupIT/beagle/tree/master/flutter)*
> - **SwiftUI and Compose:** we will try to support them in the future

Reviewer 2: ```suggestion
- **Android:** [![Maven Central](https://img.shields.io/maven-central/v/br.com.zup.beagle/android)](https://mvnrepository.com/artifact/br.com.zup.beagle/android);
- **iOS:** [![badge](https://img.shields.io/cocoapods/v/Beagle)](https://cocoapods.org/pods/Beagle)
- **React Native:** [![react native badge](https://img.shields.io/npm/v/@zup-it/beagle-react-native?logo=React)](https://www.npmjs.com/package/@zup-it/beagle-react-native);
- **Flutter:** *It is being developed, check out more information [in this folder](https://github.com/ZupIT/beagle/tree/master/flutter)*;
```
